src/tests/performance/blocks.rs

```diff
@@ -0,0 +1,222 @@
+    //  ├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
+    //  │   800 ┆      100 ┆        1 ┆      608 ┆          38 ┆      267 ┆      287 ┆      292 ┆      297 ┆      315 ┆    29.16 ┆   2743.55 │
+    //  ╰───────┴──────────┴──────────┴──────────┴─────────────┴──────────┴──────────┴──────────┴──────────┴──────────┴──────────┴───────────╯
+    //
+    //  zebra
+    //  ╭───────┬──────────┬──────────┬──────────┬─────────────┬──────────┬──────────┬──────────┬──────────┬──────────┬──────────┬───────────╮
+    //  │ Peers ┆ Requests ┆ Min (ms) ┆ Max (ms) ┆ stddev (ms) ┆ 10% (ms) ┆ 50% (ms) ┆ 75% (ms) ┆ 90% (ms) ┆ 99% (ms) ┆ Time (s) ┆ Request/s │
+    //  ╞═══════╪══════════╪══════════╪══════════╪═════════════╪══════════╪══════════╪══════════╪══════════╪══════════╪══════════╪═══════════╡
+    //  │     1 ┆      100 ┆      100 ┆      100 ┆           0 ┆      100 ┆      100 ┆      100 ┆      100 ┆      100 ┆    10.11 ┆      9.89 │
+    //  ╰───────┴──────────┴──────────┴──────────┴─────────────┴──────────┴──────────┴──────────┴──────────┴──────────┴──────────┴───────────╯
+
+    // number of requests to send per peer
+    const REQUESTS: usize = 100;
+    const REQUEST_TIMEOUT: Duration = Duration::from_millis(100);
+    // number of concurrent peers to test (zcashd hardcaps `max_peers` to 873 on my machine)
+    let peer_counts = vec![
+        1, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 200, 300, 500, 750, 800,
+    ];
+
+    let mut table = Table::new();
+    table
+        .load_preset(UTF8_FULL)
+        .apply_modifier(UTF8_ROUND_CORNERS)
+        .set_header(vec![
+            Cell::new("Peers").set_alignment(CellAlignment::Center),
+            Cell::new("Requests").set_alignment(CellAlignment::Center),
+            Cell::new("Min (ms)").set_alignment(CellAlignment::Center),
+            Cell::new("Max (ms)").set_alignment(CellAlignment::Center),
+            Cell::new("stddev (ms)").set_alignment(CellAlignment::Center),
+            Cell::new("10% (ms)").set_alignment(CellAlignment::Center),
+            Cell::new("50% (ms)").set_alignment(CellAlignment::Center),
+            Cell::new("75% (ms)").set_alignment(CellAlignment::Center),
+            Cell::new("90% (ms)").set_alignment(CellAlignment::Center),
+            Cell::new("99% (ms)").set_alignment(CellAlignment::Center),
+            Cell::new("Time (s)").set_alignment(CellAlignment::Center),
+            Cell::new("Request/s").set_alignment(CellAlignment::Center),
+        ]);
+
+    // Start node seeded with initial testnet blocks,
+    // with max peers set so that our peers should never be rejected.
+    let mut node: Node = Default::default();
+    node.initial_action(Action::SeedWithTestnetBlocks {
+        socket_addr: new_local_addr(),
+        block_count: 3,
```

Comment:
We should test larger block counts in future, 3 blocks isn't very representative of the load an actual sync would put on a node.

Reply:
Agreed.